crates/uv-workspace/src/pyproject_mut.rs

```diff
@@ -520,7 +545,9 @@ pub fn add_dependency(
         }
         // Cannot perform ambiguous updates.
         _ => Err(Error::Ambiguous),
-    }
+    };
+
+    edit
```

Comment:
We don't need to assign this to a temporary variable (i think there's a clippy lint against this?)

Reply:
> If the deps aren't sorted, we should just insert at the end
i believe the `index` points to the last position if the deps aren't sorted (one of the tests also covers this case)